src/app/ngrx/reducers/episode-performance-metrics.reducer.ts

```diff
@@ -0,0 +1,58 @@
+import * as ACTIONS from '../actions';
+
+export interface EpisodePerformanceMetricsModel {
+  seriesId: number;
+  id: number;
+  guid: string;
+  total?: number;
+  previous7days?: number;
+  this7days?: number;
+  yesterday?: number;
+  today?: number;
+  loaded?: boolean;
+  loading?: boolean;
+  error?: any;
+}
+
+export interface EpisodePerformanceMetricsState {
+  entities?: {[id: number]: EpisodePerformanceMetricsModel};
+}
+
+export const initialState = {
+  entities: {}
+};
+
+export function EpisodePerformanceMetricsReducer(state: EpisodePerformanceMetricsState = initialState, action: ACTIONS.AllActions) {
+  switch (action.type) {
+    case ACTIONS.ActionTypes.CASTLE_EPISODE_PERFORMANCE_METRICS_LOAD: {
+      const {seriesId, id, guid} = action['payload'];
+      return {
+        entities: {
+          ...state.entities,
+          [id]: {seriesId, id, guid, loading: true, loaded: false}
```

Comment:
Now that we know whether these are still loading, we can add them to the loading state and display the spinner until they're done. #221

Reply:
Is this firing twice?  When I reload the page, the metrics show, then disappear, then show up again.

http://cav.is/i/j7uct.gif

Reply:
I see what you mean about the flashing. The table component also loads the recent episode performance metrics, which on page 1 does load the same episode twice. So it updates the state once for the recent episode, then again for each of the episodes in the table. I _think_ this may be the cause of the flashing. It doesn't seem to do the flashing when the recent episode is not in the table (page > 1.)